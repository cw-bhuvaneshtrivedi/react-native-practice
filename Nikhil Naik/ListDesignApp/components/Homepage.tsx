import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
import React, { useState } from "react";
import BottomDrawer from "./BottomDrawer";

const Homepage = () => {
  const [open, setOpen] = useState(false);
  return (
    <View>
      {open ? <BottomDrawer open={setOpen} /> : null}
      <TouchableOpacity
        onPress={(e) => {
          setOpen(true);
        }}
        testID="homepageButton"
      >
        <View style={styles.button}>
          <Text style={styles.text}>Open</Text>
        </View>
      </TouchableOpacity>
    </View>
  );
};

export default Homepage;

const styles = StyleSheet.create({
  button: {
    padding: 20,
    backgroundColor: "orange",
    justifyContent: "center",
    alignItems: "center",
    width: "50%",
    alignSelf: "center",
  },
  text: { fontWeight: "bold", fontSize: 20 },
});
