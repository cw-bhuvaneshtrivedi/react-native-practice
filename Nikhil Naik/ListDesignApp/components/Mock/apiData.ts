export const apiData = [
  {
    modelId: 1729,
    modelName: "Creta",
    maskingName: "creta",
    rootId: "382",
    makeId: 8,
    makeName: "Hyundai",
    makeMaskingName: "hyundai",
    launchedOn: "01/16/2024 00:00:00",
    discontinuationDate: "",
    rootName: "Creta",
    status: "New",
    imagePath:
      "/n/cw/ec/106815/creta-exterior-right-front-three-quarter-4.png?isig=0",
    versionId: 9371,
  },
  {
    modelId: 1729,
    modelName: "Creta",
    maskingName: "creta",
    rootId: "382",
    makeId: 8,
    makeName: "Hyundai",
    makeMaskingName: "hyundai",
    launchedOn: "01/16/2024 00:00:00",
    discontinuationDate: "",
    rootName: "Creta",
    status: "New",
    imagePath:
      "/n/cw/ec/106815/creta-exterior-right-front-three-quarter-4.png?isig=0",
    versionId: 9371,
  },
  {
    modelId: 1729,
    modelName: "Ecco",
    maskingName: "Ecco",
    rootId: "382",
    makeId: 9,
    makeName: "Maruti Suzuki",
    makeMaskingName: "Maruti Suzuki",
    launchedOn: "01/16/2024 00:00:00",
    discontinuationDate: "",
    rootName: "Ecco",
    status: "New",
    imagePath:
      "/n/cw/ec/106815/creta-exterior-right-front-three-quarter-4.png?isig=0",
    versionId: 9371,
  },
  {
    modelId: 1729,
    modelName: "Ecco",
    maskingName: "Ecco",
    rootId: "382",
    makeId: 9,
    makeName: "Maruti Suzuki",
    makeMaskingName: "Maruti Suzuki",
    launchedOn: "01/16/2024 00:00:00",
    discontinuationDate: "",
    rootName: "Ecco",
    status: "New",
    imagePath:
      "/n/cw/ec/106815/creta-exterior-right-front-three-quarter-4.png?isig=0",
    versionId: 9371,
  },
];
