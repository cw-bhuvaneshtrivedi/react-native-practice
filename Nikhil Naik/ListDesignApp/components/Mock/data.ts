export const data = [
  [
    "Nikhil",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum lobortis tellus dolor. Donec sed enim fringilla, congue tortor at, venenatis nunc. Morbi in magna vitae ex iaculis vehicula non non sem. Sed euismod in nulla eget feugiat. Praesent tempor molestie leo, id iaculis lectus. Pellentesque a nulla nunc. Aliquam id.",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla mi.",
  ],
  [
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum lobortis tellus dolor. Donec sed enim fringilla, congue tortor at, venenatis nunc. Morbi in magna vitae ex iaculis vehicula non non sem. Sed euismod in nulla eget feugiat. Praesent tempor molestie leo, id iaculis lectus. Pellentesque a nulla nunc. Aliquam id.",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla mi.",
  ],
  [
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla miLorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla mi",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum lobortis tellus dolor. Donec sed enim fringilla, congue tortor at, venenatis nunc. Morbi in magna vitae ex iaculis vehicula non non sem. Sed euismod in nulla eget feugiat. Praesent tempor molestie leo, id iaculis lectus. Pellentesque a nulla nunc. Aliquam id.",
    "Lorem ipsum dolor sit amet",
  ],
  [
    "Nikhil",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum lobortis tellus dolor. Donec sed enim fringilla, congue tortor at, venenatis nunc. Morbi in magna vitae ex iaculis vehicula non non sem. Sed euismod in nulla eget feugiat. Praesent tempor molestie leo, id iaculis lectus. Pellentesque a nulla nunc. Aliquam id.",
  ],
  ["Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla mi."],
];
