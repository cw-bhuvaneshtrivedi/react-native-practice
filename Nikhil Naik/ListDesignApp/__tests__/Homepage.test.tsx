import {
  fireEvent,
  render,
  screen,
  act,
  waitFor,
} from "@testing-library/react-native";
import Homepage from "../components/Homepage";
import { apiData } from "../components/Mock/apiData";

describe("Homepage", () => {
  beforeEach(() => {
    global.fetch = jest.fn(() => Promise.resolve({ json: () => apiData }));
  });
  it("should have visible drawer", async () => {
    const result = render(<Homepage />);
    const btn = screen.getByTestId("homepageButton");
    fireEvent(btn, "click");
    waitFor(() => {
      expect(result).toMatchSnapshot();
    });
  });
  it("should have hiddden drawer", async () => {
    const result = render(<Homepage />);
    expect(result).toMatchSnapshot();
  });
});
