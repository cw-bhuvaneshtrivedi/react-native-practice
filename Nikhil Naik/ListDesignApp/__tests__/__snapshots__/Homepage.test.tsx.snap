// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should have hiddden drawer 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="homepageButton"
  >
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "orange",
          "justifyContent": "center",
          "padding": 20,
          "width": "50%",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Open
      </Text>
    </View>
  </View>
</View>
`;

exports[`Homepage should have visible drawer 1`] = `
<View>
  <View
    animatedStyle={
      {
        "value": {
          "transform": [
            {
              "translateY": 1334,
            },
          ],
        },
      }
    }
    collapsable={false}
    style={
      {
        "backgroundColor": "white",
        "height": 1334,
        "transform": [
          {
            "translateY": 1334,
          },
        ],
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "height": 120,
          "marginBottom": 4,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#484848",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Select Your Make or Model
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text />
        </View>
      </View>
      <TextInput
        placeholder="Type to Select Make "
        style={
          {
            "borderColor": "#d5d5d5",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 2,
            "fontSize": 20,
            "height": 50,
            "marginTop": 10,
            "padding": 5,
            "width": "100%",
          }
        }
      />
    </View>
    <ActivityIndicator
      size="large"
      style={
        {
          "top": "30%",
        }
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="homepageButton"
  >
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "orange",
          "justifyContent": "center",
          "padding": 20,
          "width": "50%",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Open
      </Text>
    </View>
  </View>
</View>
`;
